static/app/components/hookOrDefault.tsx

```diff
@@ -90,18 +80,20 @@ function HookOrDefault<H extends HookName>({
       }
 
       return defaultComponent;
-    },
+    }
 
     render() {
       const hookExists = this.state.hooks && this.state.hooks.length;
+      const componentFromHook = this.state.hooks[0]();
+
       const HookComponent =
-        hookExists && this.state.hooks[0]({params})
-          ? this.state.hooks[0]({params})
-          : this.getDefaultComponent();
+        hookExists && componentFromHook ? componentFromHook : this.defaultComponent;
+
+      return HookComponent ? <HookComponent {...this.props} /> : null;
```

Comment:
Can't we change `defaultComponent` to optional then?

Reply:
it already is

Reply:
yeah just saw that, lol.